src/cudamatrix/cu-sp-matrix.h

```diff
@@ -76,7 +76,7 @@ class CuSpMatrix : public CuPackedMatrix<Real> {
     CuPackedMatrix<Real>::Resize(nRows, resize_type);
   }
 
-  Real FrobeniusNorm() const { return sqrt(TraceSpSp(*this, *this)); }
+  Real FrobeniusNorm() const { cublasHandle_t handle = CuDevice::Instantiate().GetHandle(); return sqrt(TraceSpSp(handle, *this, *this)); }
```

Comment:
In situations where you use the handle only once, there is no need to declare a variable- you could just use the expression in the function call directly.  (in this case it shouldn't be needed at all once you remove it from TraceSpSp).

I suggest that you declare an inline function in cu-device.h as follows:

/// This function is declared as a more convenient way to get the CUDA device handle for use
/// in the CUBLAS v2 API, since we so frequently need to access it.
inline cublasHandle_t GetCublasHandle() {
  return CuDevice::Instantiate().GetHandle();
}
and then you can make things a bit shorter.